[TS-211] Adding users to favorites list

itsShaVawn opened the pull request:
This PR contains code and logic for adding and removing a user from the favorites list, with a maximum of 5 users in the list.

In this image you can see the function working in the console.
Starts with 3 users. Then, when you click the favorite button it adds the user to the list.
If you click the favorite button again, it removes that user from the list.

![image](https://user-images.githubusercontent.com/49250036/160534520-c7a620a9-b7a1-4192-8629-9e5d26969963.png)

kennguyen0303 commented on src/components/Profile/Profile.vue:
```diff
@@ -232,6 +242,21 @@ export default {
+    },
+    updateFavorites () {
+      if (this.list.length <= 5) {
+        if (this.favorite === false) {
+          this.favorite = !this.favorite
+          this.list = this.list.concat(this.name)
+        } else {
+          this.favorite = !this.favorite
+          this.list.splice(this.list.findIndex(item => item === this.name), 1)
+        }
```
```suggestion
// flip the status
this.favorite = !this.favorite

// update based on the status
this.favorite = this.favorite ? this.favorite.concat(this.name) : this.favorite.filter(item => item !== this.name)
```

kennguyen0303 reviewed (commented):
I just personally do not like a lot of nested if-else. Since it is my personal preference, I leave a comment here for code improvement.

itsShaVawn commented:
I addressed your comment @kennguyen0303 
I did keep splice on the array, because filter creates a new one and I want to actually change the array to remove an item.

kennguyen0303 commented:
> I addressed your comment @kennguyen0303 
> I did keep splice on the array, because filter creates a new one and I want to actually change the array to remove an item.

Thanks. In that case, I found that array.concat(..) also returns a new array. Will you want to try array.push(..) instead?

itsShaVawn commented:
> > I addressed your comment @kennguyen0303
> > I did keep splice on the array, because filter creates a new one and I want to actually change the array to remove an item.
> 
> Thanks. In that case, I found that array.concat(..) also returns a new array. Will you want to try array.push(..) instead?

i just put it in my last commit
